streaming/text/c4.py

```diff
@@ -105,3 +108,47 @@ def __getitem__(self, idx: int) -> Any:
         token_sample = self._tokenize(text_sample)
         # Skip any token grouping, currently only supporting group_method='truncate'
         return token_sample
+
+    def __iter__(self) -> Iterator[Any]:
+        """Iterable over samples.
+
+        Since concatenating samples has a custom behavior, it requires extending the
+        parent iterator class.
+
+        For `group_method = truncate`, simply return the token sample.
+        For `group_method = concat`, keep fetching token samples until it fills up the max_seq_len.
+
+        Yields:
+            Iterator[Any]: Sample iterator
+        """
+        if self.group_method == 'truncate':
+            iterator = super().__iter__()
```

Comment:
yield from super().__iter__()?

Reply:
Fixed in https://github.com/mosaicml/streaming/pull/77/commits/0a63fc68fb371e4b37fffe4ba043990b3816a9c4.